x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/support/SecurityIndexManagerTests.java

```diff
@@ -394,31 +420,31 @@ public void testCanUpdateIndexMappings() {
             getMappings(previousVersion)
         );
         manager.clusterChanged(event(markShardsAvailable(clusterStateBuilder)));
-
         manager.prepareIndexIfNeededThenExecute(prepareException::set, () -> prepareRunnableCalled.set(true));
 
         assertThat(prepareRunnableCalled.get(), is(true));
         assertThat(prepareException.get(), nullValue());
+        // Verify that the client to send put mapping was used
+        verify(threadPool, atLeastOnce()).getThreadContext();
```

Comment:
I'd prefer it if we could verify that the put mapping call was made a bit more directly, but I'm not going to block this PR if that's difficult.

Reply:
Yes, I agree. I've updated the test client to use a counter when we get a put mapping request instead. Feels a little better.